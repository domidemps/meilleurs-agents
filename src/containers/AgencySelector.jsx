/** @jsx jsx */
import {css, jsx} from '@emotion/core'
import map from 'lodash/map'
import {useDispatch, useSelector} from 'react-redux'
import {useEffect} from 'react'
import {NativeSelect} from '@material-ui/core'
import {useHistory} from 'react-router'

import {getAgencies, selectAgency} from '../actions/realtors'
import {SECONDARY_COLOR} from '../styles/material_ui_raw_theme_file'

const styles = css`
  padding: 5px;
  .options {
    background: ${SECONDARY_COLOR} !important;
    font-weight: 500 !important;
  }
}
`

export default function AgencySelector() {
  const dispatch = useDispatch()
  const agencies = useSelector(state => state.realtors.agencies)
  const selectedAgency = useSelector(state => state.realtors.selectedAgency)
  const history = useHistory()

  useEffect(() => {
    if (!Boolean(agencies)) {
      dispatch(getAgencies())
    } else {
      history.replace(`/realtor/${selectedAgency}`)
    }
  }, [agencies])

  if (!Boolean(agencies)) {
    return null
  }

  const handleSelection = event => {
    const newAgency = Number(event.target.value)
    dispatch(selectAgency(newAgency))
    history.replace(`/realtor/${newAgency}`)
  }

  const renderAgenciesChoices = agencies => {
    return map(agencies, agency => {
      return (
        <option value={agency.id} className="options" key={`agency-${agency.id}`}>
          {agency.name}
        </option>
      )
    })
  }
  return (
    <NativeSelect css={styles} onChange={e => handleSelection(e)}>
      {renderAgenciesChoices(agencies)}
    </NativeSelect>
  )
}
