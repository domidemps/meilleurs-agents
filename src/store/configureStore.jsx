import {applyMiddleware, compose, createStore} from 'redux'
import {persistReducer, persistStore} from 'redux-persist'
import storage from 'redux-persist/lib/storage'
import {routerMiddleware} from 'connected-react-router'
import promise from 'redux-promise'
import thunk from 'redux-thunk'

import reducers, {history} from '../reducers'

const middlewares = [thunk, promise, routerMiddleware(history)]

const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose

const persistConfig = {
  key: 'root',
  storage,
  blacklist: ['realtors', 'messages'],
}

const persistedRootReducer = persistReducer(persistConfig, reducers)
const createStoreWithEnhancers = composeEnhancers(applyMiddleware(...middlewares))(createStore)
export const store = createStoreWithEnhancers(persistedRootReducer)
export const persistor = persistStore(store)
