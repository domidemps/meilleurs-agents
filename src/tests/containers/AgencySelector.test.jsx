import React from 'react'
import configureStore from 'redux-mock-store'
import thunk from 'redux-thunk'
import {Provider} from 'react-redux'
import renderer from 'react-test-renderer'
import {createMemoryHistory} from 'history'
import {Router, Switch} from 'react-router-dom'

import AgencySelector from '../../containers/AgencySelector'

export const history = createMemoryHistory({initialEntries: ['/']})

describe('AgencySelector', () => {
  const store = configureStore([thunk])({
    realtors: {
      agencies: [
        {
          id: 101,
          logo: 'http://placehold.it/100x100?text=Agence+101',
          name: 'Agence #101',
          unread_messages: 73,
        },
        {
          id: 102,
          logo: 'http://placehold.it/100x100?text=Agence+102',
          name: 'Agence #102',
          unread_messages: 68,
        },
        {
          id: 103,
          logo: 'http://placehold.it/100x100?text=Agence+103',
          name: 'Agence #103',
          unread_messages: 76,
        },
      ],
      selectedAgency: '101',
    },
  })
  it('renders correctly for computers/tablets', () => {
    global.innerWidth = 600
    expect(
      renderer
        .create(
          <Provider store={store}>
            <Router history={history}>
              <Switch>
                <AgencySelector />
              </Switch>
            </Router>
          </Provider>,
        )
        .toJSON(),
    ).toMatchSnapshot()
  })
  it('renders correctly for smartphones', () => {
    global.innerWidth = 350
    expect(
      renderer
        .create(
          <Provider store={store}>
            <Router history={history}>
              <Switch>
                <AgencySelector />
              </Switch>
            </Router>
          </Provider>,
        )
        .toJSON(),
    ).toMatchSnapshot()
  })
})
