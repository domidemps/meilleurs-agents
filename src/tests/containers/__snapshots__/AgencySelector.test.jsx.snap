// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AgencySelector renders correctly for computers/tablets 1`] = `
<div
  className="MuiInputBase-root MuiInput-root MuiInput-underline css-kkkune-AgencySelector"
  onClick={[Function]}
>
  <select
    className="MuiNativeSelect-root MuiNativeSelect-select MuiInputBase-input MuiInput-input"
    onAnimationStart={[Function]}
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    value="101"
  >
    <option
      className="options"
      value={101}
    >
      Agence #101
    </option>
    <option
      className="options"
      value={102}
    >
      Agence #102
    </option>
    <option
      className="options"
      value={103}
    >
      Agence #103
    </option>
  </select>
  <svg
    aria-hidden={true}
    className="MuiSvgIcon-root MuiNativeSelect-icon"
    focusable="false"
    viewBox="0 0 24 24"
  >
    <path
      d="M7 10l5 5 5-5z"
    />
  </svg>
</div>
`;

exports[`AgencySelector renders correctly for smartphones 1`] = `
<button
  className="MuiButtonBase-root MuiIconButton-root css-hna2n5-AgencySelector"
  disabled={false}
  onBlur={[Function]}
  onClick={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={0}
  type="button"
>
  <span
    className="MuiIconButton-label"
  >
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root css-15pd8ov-AgencySelector"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M20 4H4v2h16V4zm1 10v-2l-1-5H4l-1 5v2h1v6h10v-6h4v6h2v-6h1zm-9 4H6v-4h6v4z"
      />
    </svg>
  </span>
  <span
    className="MuiTouchRipple-root"
  />
</button>
`;
